concordium-std-derive/src/lib.rs

```diff
@@ -1985,3 +1985,126 @@ pub fn concordium_cfg_test(_attr: TokenStream, item: TokenStream) -> TokenStream
     };
     out.into()
 }
+
+/// Derive the `Deletable` trait.
+///
+/// ``` ignore
+/// #[derive(Deletable)]
+/// #[concordium(state_parameter = "S")]
+/// struct MyState<S> {
+///    my_set: StateSet<MyType, S>,
+/// }
+/// ```
```

Comment:
Please provide better documentation as to **when** this needs to be derived, like we do for DeserialWithState.
Check that the generated documentation (with cargo doc for concordium-std reads well).

The documentation should state that you should only derive this **unless** you have derived Serialize, i.e., if your type has non-trivial state.

Reply:
Will expand the documentation.